sccache-dist/src/cache.rs

```diff
@@ -0,0 +1,158 @@
+use lru_disk_cache::{LruDiskCache, ReadSeek};
+use lru_disk_cache::Error as LruError;
+use lru_disk_cache::Result as LruResult;
+use ring::digest::{SHA512, Context};
+use serde_json;
+use std::collections::HashMap;
+use std::ffi::OsStr;
+use std::fs::{self, File};
+use std::io::{self, BufReader, Read, Write};
+use std::path::{Path, PathBuf};
+use std::sync::Mutex;
+
+use errors::*;
+
+// TODO: possibly shouldn't be public
+pub struct ClientToolchainCache {
+    cache_dir: PathBuf,
+    cache: Mutex<TcCache>,
+    // Local machine mapping from 'weak' hashes to strong toolchain hashes
```

Comment:
Some exposition on what weak and strong hashes are would be helpful. I gather that the strong hash is a SHA512 of the toolchain tar file.

Reply:
Added the explanation that 'weak' is just what sccache uses as part of the cache key.